homeassistant/components/stream/worker.py

```diff
@@ -0,0 +1,142 @@
+"""Proides the worker thread needed for processing streams."""
+import asyncio
+from fractions import Fraction
+import io
+
+from .core import Segment, StreamBuffer
+
+
+def generate_audio_frame():
+    """Generate a blank audio frame."""
+    from av import AudioFrame
+    audio_frame = AudioFrame(format='dbl', layout='mono', samples=1024)
+    audio_bytes = b''.join(b'\x00\x00\x00\x00\x00\x00\x00\x00'
+                           for i in range(0, 1024))
+    audio_frame.planes[0].update(audio_bytes)
+    audio_frame.sample_rate = 44100
+    audio_frame.time_base = Fraction(1, 44100)
+    return audio_frame
```

Comment:
Shall this audio_frame defined as a global variable or at least cached?

Reply:
Maybe?  It's quick to generate, and I'm not sure if using the same frame across multiple threads would cause issues.  As it stands, it only gets built once per worker thread.